skimage/measure/_find_contours.py

```diff
@@ -111,12 +117,15 @@ def find_contours(array, level,
     if array.ndim != 2:
         raise ValueError('Only 2D arrays are supported.')
     level = float(level)
+    if nodata is not None:
+        nodata = float(nodata)
```

Comment:
```suggestion
        nodata = np.double(nodata)
```
We should probably use exactly the same dtype as the `array` here.